### ISSUE TITLE
Cannot read-config from resource in uberjar

### ISSUE THREAD

**Issue by mrmcc3** [juxt/aero#15]
Hello. I'm getting this error when running an uberjar that uses `[aero "1.0.0-beta3"]`.

```
Exception in thread "main" java.lang.IllegalArgumentException: Cannot open <nil> as a Reader
```

Works fine with `[aero "1.0.0-beta2"]`

**Comment by neverfox** [comment]
Same here.

**Comment by neverfox** [comment]
This was probably introduced on [this commit](https://github.com/juxt/aero/commit/817718030e9e6545a29da247cfc6c1f92719da8e#diff-1fef30a8e2221a6f4c3d14bb67e8dc34) and could be related to the warning given [here](https://clojuredocs.org/clojure.java.io/resource):

> If you need to slurp a file from a JAR file, don't call io/file on the result of calling io/resource, or you will get an exception that says the resource is "not a file". Instead, call slurp directly on the result of io/resource.

The new commit makes use of `io/file`, which works fine outside an uberjar (and probably passes tests). Will have to revert to beta2 in the meantime.

**Comment by malcolmsparks** [comment]
Thanks for this - it's noted. We need to rethink this and implement it in a way that fits uberjars. To the hammock!

**Comment by weavejester** [comment]
One solution would be to expose the include paths as an option, rather than trying to resolve them automatically. For example:

``` clojure
(read-config "dev/config.edn" {:includes {"webserver.edn" "dev/webserver.edn"}})
```

Naturally you could also use URLs and other sources that `read-config` supports:

``` clojure
(read-config "dev/config.edn" {:includes {"webserver.edn" (io/resource "common/webserver.edn")}})
```

You could also use a function instead of a map:

``` clojure
(read-config "dev/config.edn {:includes (fn [path] (io/file "dev" path))})
```

This also allows configurations to be ignorant of where includes come from. For instance, you could write:

``` edn
{:webserver #include :webserver
 :analytics #include :analytics}
```

And load it as:

``` clojure
(read-config "dev/config.edn" {:includes {:webserver "dev/webserver.edn"
                                          :analytics "dev/analytics.edn"}})
```

**Comment by malcolmsparks** [comment]
I like this idea. 

It strongly reminds me of the XML entity resolvers in Java: http://www.saxproject.org/apidoc/org/xml/sax/EntityResolver.html, which was basically a way of doing the same thing - if you see a URL like this, load it this way.

I think the feature can be implemented in such a way that such 'entity resolvers' can be added optionally for those who need the extra flexibility.

**Comment by mrmcc3** [comment]
fixed in 1.0.0-beta5
